awx/api/serializers.py

```diff
@@ -732,7 +736,25 @@ class EmptySerializer(serializers.Serializer):
     pass
 
 
-class UnifiedJobTemplateSerializer(BaseSerializer):
+class OpaQueryPathEnabledMixin(serializers.Serializer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not flag_enabled("FEATURE_POLICY_AS_CODE_ENABLED") and 'opa_query_path' in self.fields:
+            self.fields.pop('opa_query_path')
```

Comment:
I'm still so incredibly worried about this from a performance and consistency standpoint, and I hope we are able to make this non-conditional soon.

Reply:
kinda agree...

Reply:
but it's the pattern for APIs behind feature flag ain't it ?

Reply:
Like RBAC, there may be a good argument to have OPA API structure at arms-length (having its own section of the API) from the rest of the API. Inter-mingling the things causes problems like these. A flag that makes an entire view accessible or not accessible could be more reasonably turned off by a feature flag, safely tucked away at a view-level override, and honest documentation and OpenAPI specs.

Reply:
To put views/APIs behind feature flags, I'd recommend we check out these options - 
URLS - https://cfpb.github.io/django-flags/api/urls/
Views - https://cfpb.github.io/django-flags/api/decorators/

This code was removed a while ago, but initially this is how we gated an API endpoint behind feature flags - https://github.com/ansible/eda-server/pull/1132/files#diff-2970751c57bcb2f4906a6353ed7d8f16c396995e16c41b35bf6c84a780c5b4cfR110-R120

Reply:
Good to know. The `flagged_path` is the reference implementation we should look to.

https://github.com/cfpb/django-flags/blob/677e2bfdfe22ecf0d7bfb87119999bb91ce8085c/flags/urls.py#L123

That uses a decorator, so the view is still the view.

https://github.com/cfpb/django-flags/blob/main/flags/decorators.py#L46

But as a runtime thing it does `raise Http404`

So this explains how it does something that _appears would check a flag at import_ and delays it to runtime, as we would want.